integration-provisioning/src/main/java/org/symphonyoss/integration/provisioning/service/ApplicationService.java

```diff
@@ -77,22 +79,30 @@ public void init() {
 
   /**
    * Creates or updates an application on Symphony store.
-   * @param configurationId The identifier for the integration configuration associated with the
-   * application.
+   * @param settings Integration settings associated with the application.
    * @param application Application object.
    */
-  public void setupApplication(String configurationId, Application application) {
+  public void setupApplication(IntegrationSettings settings, Application application) {
     String appType = application.getComponent();
 
     LOGGER.info("Provisioning Symphony store data for application: {}", appType);
 
-    User user = userService.getUser(appType);
-    String botUserId = user.getId().toString();
+    if (settings.getOwner() == null) {
+      User user = userService.getUser(appType);
```

Comment:
@ecarrenho-daitan I'm assuming the new version of Integration API will return the integration owner. Then, this if clause would be used to the legacy version of the API, therefore I need to use the app type as the integration username.